### ISSUE TITLE
Problem of relative data's accuracy

### ISSUE THREAD

**Issue by NekoDisc** [SIMRacingApps/SIMRacingApps#197]
SIMRacingApps's relative data has not enough accuracy.
In the worst case, it is near twice as time as ingame blackbox.
I hope that SIMRacingApps has more correctly relative data.

![relative](https://user-images.githubusercontent.com/48743794/54657968-c0ee9f80-4b0e-11e9-9bfe-43083c336c65.jpg)

**Comment by jfgilliam** [comment]
I agree it could be improved. In its current state it simply calculates how long it would take at your current speed to reach the point on the track where the other car is. Of course, that is not reality because you will either be accelerating or decelerating most of the time. This is worse on road courses than it is on ovals. 

The only way to improve on this is to divide the track up into many segments and keep track of the time spent in each segment and then use that to project your time. This has its own flaws. Do you use your best time or an average, etc. 

I’m not even sure how iRacing does it. But I will not be able to match them. Knowing this, your feedback is welcome. 

- Jeff

**Comment by NekoDisc** [comment]
I have an idea that perhaps relative's accuracy improve than now, but I have no confidence to explaining well in English.
However, I will try anyway.

1. Make the checkpoints based on a distance from the control line.
Divide the track every N meters and numbering it.
2. Record the session time(not lap time) when passing every checkpoints about every cars.
3. Compare a latest checkpoint time which has same number based on the car behind.
4. Perhaps this logic need to override the checkpoint time which the car already passed If the car too slow.

Example:
![relative](https://user-images.githubusercontent.com/48743794/54731442-91ed3200-4bd1-11e9-8a03-ce33e22d8c7c.jpg)
Car3 running at 310m
Car1-3, Use check point 3, 0.17
Car2-3, Use check point 3, 0.08
Car3-4, Use check point 2, -0.05
Car3-5, Use check point 2, -0.10

This idea has the problem that the large track like the Nordschleife need too many checkpoints.
